core/src/main/java/com/sequenceiq/cloudbreak/service/image/ImageCatalogService.java

```diff
@@ -317,12 +319,41 @@ public StatedImages getImages(Long workspaceId, String name, String provider) th
     public StatedImages getImages(Long workspaceId, String name, Set<String> providers) throws CloudbreakImageCatalogException {
         try {
             ImageCatalog imageCatalog = get(workspaceId, name);
-            return getImages(new ImageFilter(imageCatalog, providers, cbVersion, baseImageEnabled(), null, null));
+            if (Strings.isNullOrEmpty(imageCatalog.getImageCatalogUrl()) && imageCatalog.getCustomImages() != null) {
```

Comment:
Conditions like this that clearly identify something, IMHO could be extracted into a function, even if only called a few times. Something like isCustomCatalog(imageCatalog) looks more readable.

Reply:
ACK